File: src/main/java/quebec/virtualite/backend/services/utils/CustomExceptionHandlerResolver.java
package quebec.virtualite.backend.services.utils;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.HttpRequestMethodNotSupportedException;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

import javax.validation.ConstraintViolationException;

import static org.springframework.http.HttpStatus.BAD_REQUEST;

@Component
@ControllerAdvice
@Slf4j
public class CustomExceptionHandlerResolver
{
    @ExceptionHandler({
        ConstraintViolationException.class,
        HttpRequestMethodNotSupportedException.class,
        MethodArgumentNotValidException.class
    })
    public ResponseEntity<String> customExceptionHandler(Exception e)
    {
        log.warn(e.getMessage());
        return new ResponseEntity<>(BAD_REQUEST);
    }
}

File: src/main/java/quebec/virtualite/backend/services/utils/CustomExceptionHandlerResolver.kt
package quebec.virtualite.backend.services.utils

import org.springframework.http.HttpStatus.BAD_REQUEST
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component
import org.springframework.web.HttpRequestMethodNotSupportedException
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import javax.validation.ConstraintViolationException

@Component
@ControllerAdvice
class CustomExceptionHandlerResolver
{
    @ExceptionHandler(
        ConstraintViolationException::class,
        HttpRequestMethodNotSupportedException::class,
        MethodArgumentNotValidException::class,
    )
    fun customExceptionHandler(e: Exception): ResponseEntity<String>
    {
        return ResponseEntity(BAD_REQUEST)
    }
}